services/workflow/workflow-service/src/main/java/com/milesight/beaveriot/rule/components/entityselector/EntitySelectorComponent.java

```diff
@@ -0,0 +1,42 @@
+package com.milesight.beaveriot.rule.components.entityselector;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.milesight.beaveriot.context.api.EntityValueServiceProvider;
+import com.milesight.beaveriot.context.integration.model.ExchangePayload;
+import com.milesight.beaveriot.rule.annotations.OutputArguments;
+import com.milesight.beaveriot.rule.annotations.RuleNode;
+import com.milesight.beaveriot.rule.api.ProcessorNode;
+import com.milesight.beaveriot.rule.constants.RuleNodeType;
+import com.milesight.beaveriot.rule.support.SpELExpressionHelper;
+import lombok.Data;
+import org.apache.camel.Exchange;
+import org.apache.camel.spi.UriParam;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author loong
+ * @date 2024/12/18 8:47
+ */
+@RuleNode(value = "entitySelector", type = RuleNodeType.ACTION, description = "EntitySelector")
+@Data
+public class EntitySelectorComponent implements ProcessorNode<Exchange> {
+
+    @OutputArguments
+    @UriParam(javaType = "java.util.List")
+    private List<String> entities;
+
+    @Autowired
+    EntityValueServiceProvider entityValueServiceProvider;
+
+    @Override
+    public void processor(Exchange exchange) {
+        List<String> entitiesVariables = (List<String>) SpELExpressionHelper.resolveExpression(exchange, entities);
+        Map<String, JsonNode> entityValues = entityValueServiceProvider.findValuesByKeys(entitiesVariables);
+        Map<String, Object> entityValuesMap = entityValues.entrySet().stream().collect(java.util.stream.Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
```

Comment:
The object should be actual object value, not the JsonNode

Reply:
fixed